apps/snitch_core/lib/core/domain/order/order.ex

```diff
@@ -38,7 +38,7 @@ defmodule Snitch.Domain.Order do
   """
   @spec payments_total(Order.t(), String.t()) :: Money.t()
   def payments_total(order, payment_state) do
-    {:ok, currency} = Defaults.fetch(:currency)
+    currency = GCModel.fetch_currency()
```

Comment:
I would really prefer to have `{:ok, currency}` or `{:error, msg}` rather than returning just as currency. this way matching becomes easier. @SagarKarwande thoughts?

Reply:
We can use just the return value since we will be using a `Repo.one` query, which either gives a value or nil.

Reply:
Yes returning tuples with success and error will help.

@ashish173 As per the definition of  `GeneralConfiguration.fetch_currency()` we would never be able to get the error case.

Reply:
That's okay @SagarKarwande for now. Since we always start/provision a store with a currency. I think now I agree with what @pkrawat1 is saying.

Reply:
But there would be no {:error, msg} scenario, it will always return a string.